### PR TITLE
adds a new 'nullable' validation option which allows null values as validated data

### PR DESCRIPTION
**Pull request by lordrhodos**
This PR overcomes the limitation to explicitly validate and accept an optional data key set to `null`. At the moment any value without a validation rule `required` or `nonEmpty` set to `null` will not be set in the internal `validatedData` array.

The PR adds a new virtual `nullable` configuration option to the Validator class allowing a `null` value to be set in the internal `validatedData` array.

### CODE REVIEW

File: tests/ValidatorTest.php [hunk 1]
```diff
@@ -338,4 +339,22 @@ public function testGetValidRuleProperties()
     {
         $this->assertContains('required', $this->validator->getValidRuleProperties());
     }
+
+    public function testValidateNullableValueDoesNotReturnErrorMsg()
```
Comment (felixsand) [inline tests/ValidatorTest.php]: Also add tests for sending in other data here - like strings, `0`, empty strings, etc to make sure this always work

File: src/Validator.php [hunk 2]
```diff
@@ -126,7 +127,10 @@ public function validate($data)
                 if (is_string($value) && (!isset($ruleProperties['trim']) || $ruleProperties['trim'] === true)) {
                     $value = trim($value);
                 }
-            // Don't validate empty values that are not set and not required
+                // allow a nullable value to be added as validated data
+            } elseif (!empty($ruleProperties['nullable']) && array_key_exists($fieldName, $data)) {
+                $value = null;
```
Comment (felixsand) [inline src/Validator.php:132]: Will this not set _all_ $values with `nullable => true` to `null`? 🤔 
So for example - if I send in `foo-bar` to a field with validator rule `nullable => true` that fields value will always be set to `null`

**Review by felixsand — CHANGES_REQUESTED** [review]
A couple of questions 🙂

File: src/Validator.php (hunk 2 above)
Comment (lordrhodos) [reply src/Validator.php:132]: no, that case is covered in the first if clause

File: tests/ValidatorTest.php (hunk 1 above)
Comment (lordrhodos) [reply tests/ValidatorTest.php]: good idea!

**Review by felixsand — APPROVED** [review]
I think all looks good now! :-)
